### PR TITLE
Listener new

### PR DESCRIPTION
**Pull request by vibhuti910**

### CODE REVIEW

File: src/mtool/api/UnitTests/Subsystem/test_subsystem.py [hunk 1]
```diff
@@ -54,6 +54,14 @@
         "target_address":"111.100.13.97",
         "transport_service_id":"1158"
 }'''
+listener_delete_json = '''{
+    "data": {
+        "subnqn": "nqn.2019-04.pos:subsystem1",
+        "transportType": "TCP",
+        "ip": "107.108.83.97",
```
Comment (manoj22-k) [inline src/mtool/api/UnitTests/Subsystem/test_subsystem.py:61]: Please use 127.0.0.1 here

File: src/mtool/api/rest/app.py [hunk 2]
```diff
@@ -127,4 +127,4 @@ def handle_invalid_usage(error):
         host='0.0.0.0',
         debug=False,
         use_reloader=False,
-        port=5000)
\ No newline at end of file
+        port=5005)
```
Comment (manoj22-k) [inline src/mtool/api/rest/app.py:130]: change port to 5000

File: src/mtool/api/rest/blueprints/subsystem.py [hunk 3]
```diff
@@ -93,4 +93,23 @@ def add_listener(current_user):
...
+    print("delete listener is called")
+    try:
+        body_unicode = request.data.decode('utf-8')
+        body = json.loads(body_unicode)
+        data = body.get('data')
+        resp = dagent.delete_listener(data)
+        if resp is not None:
+            resp = resp.json()
+            return toJson(resp)
+        else:
```
Comment (manoj22-k) [inline src/mtool/api/rest/blueprints/subsystem.py:111]: Avoid else

File: src/mtool/api/rest/app.py (hunk 2 above)
Comment (vishal-shakya-ssir) [inline src/mtool/api/rest/app.py:130]: Please change the port to 5000

File: src/mtool/api/rest/rest_api/dagent/ibofos.py [hunk 4]
```diff
@@ -368,6 +368,61 @@ def array_exists(arrayname=array_names[0],auth=BASIC_AUTH_TOKEN):
...
 
+def delete_subsystem(name, auth=BASIC_AUTH_TOKEN):
+    logger = logging.getLogger(__name__)
+    logger.info('%s', 'Sending command to D-Agent to delete subsystem api ...')
+    req_headers = get_headers(auth)
+    request_body = {
+        "param": {
+            "subnqn": name}}
+    request_body = json.dumps(request_body)
+    print(request_body)
```
Comment (manoj22-k) [inline src/mtool/api/rest/rest_api/dagent/ibofos.py:379]: remove this print statement

File: src/mtool/api/rest/rest_api/dagent/ibofos.py [hunk 5]
```diff
@@ -368,6 +368,61 @@ def array_exists(arrayname=array_names[0],auth=BASIC_AUTH_TOKEN):
...
+            url= BASE_URL +
+            'subsystem',
+            headers=req_headers,
+            timeout=(
+                connect_timeout,
+                read_timeout),
+            data=request_body)
+        return response
+    except Exception as err:
+        print(f'Other error occurred: {err}')
```
Comment (manoj22-k) [inline src/mtool/api/rest/rest_api/dagent/ibofos.py:392]: remove this print statement

File: src/mtool/ui/package.json [hunk 6]
```diff
@@ -94,7 +94,7 @@
     "lint-staged": "^9.5.0",
     "prettier": "^1.19.1"
   },
-  "proxy": "http://localhost:5000",
+  "proxy": "http://localhost:5005",
```
Comment (vishal-shakya-ssir) [inline src/mtool/ui/package.json:97]: same as above comment

File: src/mtool/api/rest/rest_api/dagent/ibofos.py [hunk 7]
```diff
@@ -368,6 +368,61 @@ def array_exists(arrayname=array_names[0],auth=BASIC_AUTH_TOKEN):
...
+    return make_failure_response(
```
Comment (manoj22-k) [inline src/mtool/api/rest/rest_api/dagent/ibofos.py:393]: This return statement should be in exception block

File: src/mtool/ui/package.json (hunk 6 above)
Comment (manoj22-k) [inline src/mtool/ui/package.json:97]: change port to 5000

File: src/mtool/ui/src/components/SubsystemDetails/index.js [hunk 8]
```diff
@@ -31,8 +31,8 @@
  */
 
 import React, { useState } from 'react';
-import { Grid, Typography, withStyles, ThemeProvider } from '@material-ui/core';
-import { Add, Check, Clear, FirstPage, LastPage, Search, ChevronRight, ChevronLeft, Remove, ArrowUpward } from '@material-ui/icons';
+import { Grid, Typography, withStyles, ThemeProvider, Tooltip,IconButton} from '@material-ui/core';
+import { Add, Check, Clear, FirstPage, LastPage, Search, ChevronRight, ChevronLeft, Remove, ArrowUpward,  Delete} from '@material-ui/icons';
```
Comment (manoj22-k) [inline src/mtool/ui/src/components/SubsystemDetails/index.js:35]: Multiple spaces before Delete

File: src/mtool/ui/src/sagas/subsystemSaga.js [hunk 9]
```diff
@@ -237,7 +237,59 @@ export function* createSubsystem(action) {
...
+        yield put(actionCreators.showSubsystemAlert({
+          ...alertDetails,
+          errorCode: `Description: ${response.data && response.data.result && response.data.result.status
+            ? `${response.data.result.status.description}`
+            : "Agent Communication Error"
+            }`
+        }));
+      }
+    } catch (error) {
+      console.log("Error Occured",error)
```
Comment (manoj22-k) [inline src/mtool/ui/src/sagas/subsystemSaga.js:288]: remove console log

File: stackhawk.yml [hunk 10]
```diff
@@ -0,0 +1,4 @@
+app:
+  applicationId: 2bba9e38-db0b-4701-93ba-66d61d230cf3 # (required)
+  env: Development # (required)
+  host: http://localhost:3000 # (required)
```
Comment (manoj22-k) [inline stackhawk.yml:4]: Please remove this file

**Review by vishal-shakya-ssir — APPROVED** [review]
LGTM, Please check the minor comment 
Thanks

File: src/mtool/api/rest/blueprints/subsystem.py [hunk 11]
```diff
@@ -93,4 +93,23 @@ def add_listener(current_user):
...
         print("Exception in adding listener " + e)
-        return abort(404)
\ No newline at end of file
+        return abort(404)
+    
+#delete listner
+@subsystem_bp.route('/api/v1/listener/', methods=['DELETE'])
+@token_required
+def delete_listener(current_user):
+    print("delete listener is called")
```
Comment (manoj22-k) [inline src/mtool/api/rest/blueprints/subsystem.py:102]: Remove this print statement

File: src/mtool/api/rest/rest_api/dagent/ibofos.py [hunk 12]
```diff
@@ -368,6 +368,59 @@ def array_exists(arrayname=array_names[0],auth=BASIC_AUTH_TOKEN):
...
+    request_body = {
+        "param": {
+        "subnqn": data['subnqn'],
+        "transportType": data['transportType'],
+        "targetAddress": data['ip'],
+        "transportServiceId": data['port']
+    }
+}
+    request_body = json.dumps(request_body)
+    print(request_body)
```
Comment (manoj22-k) [inline src/mtool/api/rest/rest_api/dagent/ibofos.py:407]: Remove print statements

**Pull request merged**